### ISSUE TITLE
NPU+deepspeed zero3环境下--max_grad_norm失效

### ISSUE THREAD

**Issue by SeuZL** [modelscope/ms-swift#4198]
**Describe the bug**
What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图)

我才用了以下的训练代码：
nohup bash -c 'export ASCEND_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 HCCL_CHECK_TIMEOUT=600 MAX_PIXELS=802816 TORCH_NPU_FUSION_ENABLE=1; torchrun --nproc_per_node=8 --rdzv_conf "overlap_timeout=600" /root/zhangliang/Qwen2.5-VL/Lora/ms-swift/swift/cli/sft.py --ddp_backend hccl --model /root/nfs_storage/zhangliang/Qwen2.5vl-32B-Instruct --dataset /root/zhangliang/Qwen2.5-VL/yanke/filtered_data_imagetoend.jsonl --deepspeed /root/zhangliang/Qwen2.5-VL/yanke/deepspeedzero3.json --train_type lora --torch_dtype bfloat16 --num_train_epochs 1 --per_device_train_batch_size 1 --gradient_accumulation_steps 8 --learning_rate 1e-5 --lr_scheduler_type cosine --warmup_ratio 0.1 --weight_decay 0.01 --lora_rank 32 --lora_alpha 64 --lora_dropout 0.1 --use_rslora False --max_grad_norm 1.0 --target_modules all-linear --freeze_vit true --eval_steps 300 --save_steps 300 --save_total_limit 50 --logging_steps 30 --max_length 8192 --output_dir /data/output/yanke --dataloader_num_workers 8 --model_kwargs "{\"device_map\":{\"\": \"npu:auto\"}}" --optim adamw_torch_npu_fused --gradient_checkpointing True' > /data/logs/qwen2.5lora/alldata32B2.txt 2>&1 &
其中已经设置了
--max_grad_norm 1.0
但是训练过程中，这一条明显不生效
                                                                
{'loss': 3.66410777, 'token_acc': 0.35908569, 'grad_norm': 14.55122662, 'learning_rate': 9.93e-06, 'train_speed(iter/s)': 0.011751, 'epoch': 0.3, 'global_step/max_steps': '870/5826', 'percentage': '14.93%', 'elapsed_time': '20h 33m 48s', 'remaining_time': '4d 21h 8m 26s'}

**Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等)

使用的是昇腾架构，8张NPU，模型是qwen2.5vl32B

**Additional context**
Add any other context about the problem here(在这里补充其他信息)

**Comment by Jintao-Huang** [comment]
https://github.com/huggingface/transformers/issues/37566
